ADR/0002-ca-generation-with-constraints.md

```diff
@@ -0,0 +1,155 @@
+# 2. CA generation with security constraints
+
+Date: 2025-11-09
+
+## Status
+
+Accepted
+
+## Context
+
+The Squid proxy uses SSL bumping to intercept HTTPS traffic, requiring a Certificate
+Authority (CA) to sign dynamically generated certificates. The primary security risk is
+CA private key compromise, which would allow an attacker to sign certificates for any
+domain.
+
+To mitigate this risk, we need to apply X.509 certificate constraints:
+- **pathLenConstraint=0**: Prevents creation of intermediate CAs, forcing direct signing
+  by the root CA
+- **Name Constraints**: Limits which domains the CA can issue certificates for
+- **Key Usage**: Restricts certificate to CA operations only (cert sign, CRL sign)
+
+cert-manager cannot generate CAs with `pathLenConstraint=0`.
+
+## Decision
+
+We will generate the root CA certificate externally using OpenSSL with the required
+constraints, then distribute it via Kubernetes Secrets and trust-manager.
+
+**CA Generation:**
+- Generate CA with OpenSSL using a configuration file specifying:
+  - `pathLenConstraint=0` (Basic Constraints)
+  - Name Constraints (permitted: `registry.access.redhat.com`, `registry.redhat.io`,
+    `*.redhat.com`, `*.access.redhat.com`, `*.connect.redhat.com`, `quay.io`,
+    `.quay.io`, `cdn.quay.io`, `cdn01.quay.io`, `cdn02.quay.io`; excluded: `*`)
+  - Key Usage (`keyCertSign`, `cRLSign` only)
+  - 90-day validity period
```

Comment:
Is this a 90 day validity for the CA cert or those issued from it? Any reason for picking 90 days?

Reply:
I changed it to TBD.